app/models/concerns/event_concern.rb

```diff
@@ -0,0 +1,14 @@
+module EventConcern
+  extend ActiveSupport::Concern
+
+  included do
+    after_destroy :raise_event, :prepend => true
```

Comment:
This is cool, I had no idea about `:prepend => true` to make sure the dependents are deleted before the parent.